src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php

```diff
@@ -590,9 +590,7 @@ private function validateClassNode($object, $cacheKey, ClassMetadataInterface $m
                     $object,
                     $cacheKey.':'.$propertyName,
                     $propertyMetadata,
-                    $propertyPath
-                        ? $propertyPath.'.'.$propertyName
-                        : $propertyName,
+                    PropertyPath::append($propertyPath, $propertyName),
```

Comment:
I don't think we need another unit test here. The two cases of the current logic are (or at least should be) already tested. Exchanging the current explicit logic by a method call should not require additional tests, as we don't change the functionality. The internal behavior of `append()` is tested elsewhere.